### PR TITLE
HOSTEDCP-1030: Add destroy cluster for Agent for HCP CLI

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Add destroy cluster for Agent for HCP CLI

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1030](https://issues.redhat.com/browse/HOSTEDCP-1030)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: product-cli/cmd/cluster/agent/destroy.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+	"github.com/openshift/hypershift/cmd/cluster/agent"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	"github.com/openshift/hypershift/cmd/log"
+)
+
+func NewDestroyCommand(opts *core.DestroyOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "agent",
+		Short:        "Destroys a HostedCluster and its associated infrastructure on Agent.",
```
Comment (enxebre) [inline product-cli/cmd/cluster/agent/destroy.go:14]: I don't see we using period of the sentences in "Short" here https://github.com/openshift/hypershift/pull/2755/files#diff-a0b6865b7f8d3556ea1be8721258c0d1ff94f8c4b773b8ceb1947feef238e974R13
can we be consistent?

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**